lib/auth/integration/integrationv1/awsra.go

```diff
@@ -48,14 +52,14 @@ func (s *Service) GenerateAWSRACredentials(ctx context.Context, req *integration
 
 // generateAWSRACredentialsWithoutAuthZ generates a set of AWS credentials which uses the AWS Roles Anywhere integration.
 // Bypasses authz and should only be used by other methods that validate AuthZ.
-func (s *Service) generateAWSRACredentialsWithoutAuthZ(ctx context.Context, req *integrationpb.GenerateAWSRACredentialsRequest) (*integrationpb.GenerateAWSRACredentialsResponse, error) {
-	integration, err := s.cache.GetIntegration(ctx, req.GetIntegration())
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
-	spec := integration.GetAWSRolesAnywhereIntegrationSpec()
-	if spec == nil {
-		return nil, trace.BadParameter("integration %q is not an AWSRA integration", req.Integration)
+func (s *Service) generateAWSRACredentialsWithoutAuthZ(ctx context.Context, req *integrationpb.GenerateAWSRACredentialsRequest, trustAnchor string) (*integrationpb.GenerateAWSRACredentialsResponse, error) {
+	if trustAnchor == "" {
+		spec, err := s.getAWSRolesAnywhereIntegrationSpec(ctx, req.GetIntegration())
+		if err != nil {
+			return nil, trace.Wrap(err)
```

Comment:
Add a message to this error? Maybe _trust anchor was not provided and could not be fetched from the integration_.